code/controllers/subsystem/dbcore.dm

```diff
@@ -264,26 +264,39 @@ SUBSYSTEM_DEF(dbcore)
   * * querys - List of queries to execute
   * * warn - Boolean to warn on query failure
   * * qdel - Boolean to enable auto qdel of queries
+  * * assoc - Boolean to enable support for an associative list of queries
   */
-/datum/controller/subsystem/dbcore/proc/MassExecute(list/querys, warn = FALSE, qdel = FALSE)
+/datum/controller/subsystem/dbcore/proc/MassExecute(list/querys, warn = FALSE, qdel = FALSE, assoc = FALSE)
 	if(!islist(querys))
 		if(!istype(querys, /datum/db_query))
 			CRASH("Invalid query passed to MassExecute: [querys]")
 		querys = list(querys)
 
+	var/start_time = start_watch()
+	log_debug("Mass executing [length(querys)] queries...")
+
 	for(var/thing in querys)
-		var/datum/db_query/query = thing
+		var/datum/db_query/query
+		if(assoc)
```

Comment:
The `assoc` var is unnecessary. You can determine whether the list is associative here. 

I'm not sure what the best way to go about it is, but you can probably write some simple generic helper that will do that. Coderbus says an option is checking if the list's json conversion starts with a curly bracket, but I somehow doubt that that's the best way to do it.

Reply:
Json converting an entire list just to check if its associative or not is a horrible way to do this, and is not being done. There a reason we have `QDEL_LIST`, `QDEL_LIST_ASSOC` and `QDEL_LIST_ASSOC_VALUE`, instead of code to assume what kind of list it is.